### PR TITLE
[zgui][BREAKING] Update ImGui + test engine to 1.90.4-docking and ImPlot 0.17

### PR DESCRIPTION
**Pull request by OndraVoves**
Thanks @dstrachan for initial work in #407 👍 
Btw problem with tiny scrollbar is not probably bug. In example size is set to 6 but in imgui default is 14, but maybe i miss something.

### beginChild and beginChildId
Change
```zig
zgui.beginChild("FOO", .{ .border = true }))
```
To
```zig
zgui.beginChild("FOO",.{ .child_flags = .{ .border = true } })
```

### CODE REVIEW

**Comment by hazeycode** [comment]
Nice work! I'll review this properly ASAP.

**Review by hazeycode — COMMENTED** [review]
I think we want an `enable_docking` option that defines `IMGUI_HAS_DOCK` also? Default to true?

File: samples/gui_test_wgpu/src/gui_test_wgpu.zig [hunk 1]
```diff
@@ -106,7 +106,7 @@ fn create(allocator: std.mem.Allocator, window: *zglfw.Window) !*DemoState {
 
     style.window_min_size = .{ 320.0, 240.0 };
     style.window_border_size = 8.0;
-    style.scrollbar_size = 6.0;
+    style.scrollbar_size = 12.0;
```
Comment (hazeycode) [inline samples/gui_test_wgpu/src/gui_test_wgpu.zig:109]: I'm nervous about this because I don't understand why it's changed.

Comment (OndraVoves) [reply samples/gui_test_wgpu/src/gui_test_wgpu.zig]: FIxed. My bad ;)

Comment (hazeycode) [reply samples/gui_test_wgpu/src/gui_test_wgpu.zig]: But without this change, the scroll bar is thinner than before

**Comment by OndraVoves** [comment]
> I think we want an `enable_docking` option that defines `IMGUI_HAS_DOCK` also? Default to true?

On docking branch is defined by imgui itself and it work like normal imgui until you  enable "dock feeature"  by `dock_enable` in `zgui.ConfigFlags`.  (define `IMGUI_HAS_DOCK ` in build.zig is my mistake)

File: samples/gui_test_wgpu/src/gui_test_wgpu.zig (hunk 1 above)
Comment (OndraVoves) [reply samples/gui_test_wgpu/src/gui_test_wgpu.zig]: Yes because 6 is really thin, default in imgui is `12`. strange i know 😅

**Comment by hazeycode** [comment]
> define `IMGUI_HAS_DOCK ` in build.zig is my mistake

I see `IMGUI_HAS_DOCK` used by preprocessor in imgui.cpp. I think we want to also compile that code?

**Comment by OndraVoves** [comment]
> > define `IMGUI_HAS_DOCK ` in build.zig is my mistake
> 
> I see `IMGUI_HAS_DOCK` used by preprocessor in imgui.cpp. I think we want to also compile that code?

Corect and its defined here https://github.com/zig-gamedev/zig-gamedev/blob/ca91e27873a237a056287974cf487b4ed6c3f85b/libs/zgui/libs/imgui/imgui.h#L30

File: samples/gui_test_wgpu/src/gui_test_wgpu.zig (hunk 1 above)
Comment (hazeycode) [reply samples/gui_test_wgpu/src/gui_test_wgpu.zig]: It seems about right. Maybe something was fixed upstream. Let's go with it.

**Comment by OndraVoves** [comment]
🎉 NICE 🎉

**Comment by Pyrolistical** [comment]
Hmm, this change broke my local wip with:
```sh
Assertion failed: viewport->RendererUserData == 0 && viewport->PlatformUserData == 0, file ...\libs\zgui\libs\imgui\imgui.cpp, line 15466
```

I am doing something out of the ordinary by re-initing zgui.

**Pull request merged**